File: infobip-mobile-messaging-android-chat-sdk/src/main/java/org/infobip/mobile/messaging/chat/core/InAppChatWebViewManager.java
package org.infobip.mobile.messaging.chat.core;

/**
 * Interface for Views with lifecycle (Activity, Fragment) that manage WebView
 */
public interface InAppChatWebViewManager {

    void onPageStarted();
    void onPageFinished();
    void setControlsEnabled(boolean enabled);
    void onJSError(String message);
    void setControlsVisibility(boolean isVisible);
    void openAttachmentPreview(String url, String type, String caption);
    void setLanguage(String language);
    void onWidgetViewChanged(InAppChatWidgetView widgetView);
    void sendContextualMetaData(String data, InAppChatMultiThreadFlag multiThreadFlag);
}

File: infobip-mobile-messaging-android-chat-sdk/src/main/java/org/infobip/mobile/messaging/chat/core/InAppChatWebViewManager.kt
package org.infobip.mobile.messaging.chat.core

/**
 * Interface for Views with lifecycle (Activity, Fragment) that manage WebView
 */
internal interface InAppChatWebViewManager {
    fun onPageStarted(url: String)
    fun onPageFinished(url: String)
    fun onWidgetApiError(method: InAppChatWidgetApiMethod, errorPayload: String?)
    fun onWidgetApiSuccess(method: InAppChatWidgetApiMethod, successPayload: String?)
    fun setControlsVisibility(isVisible: Boolean)
    fun openAttachmentPreview(url: String?, type: String?, caption: String?)
    fun onWidgetViewChanged(widgetView: InAppChatWidgetView)
    fun onWidgetRawMessageReceived(message: String?)
}